server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -378,7 +342,7 @@ protected final void failIfNoDocValues() {
     }
 
     protected final void failIfNotIndexed() {
-        if (indexOptions() == IndexOptions.NONE && pointDimensionCount() == 0) {
+        if (isSearchable == false) {
```

Comment:
I find this confusing given that soon we'll have searchable fields that are not indexed. Should we rather rename it to `failIfNotSearchable` ? And the error message? Or maybe isSearchable should rather be isIndexed ?

Reply:
I'll change the various to be `isIndexed` - I think the `isSearchable` method should stay named as it is though?

Reply:
yes the existing method that is used by field_caps , I would leave unchanged